zk/tx/tx.go

```diff
@@ -189,6 +189,8 @@ func DecodeBatchL2Blocks(txsData []byte, forkID uint64) ([]DecodedBatchL2Data, e
 	return result, nil
 }
 
+type TxDecoder func([]byte, byte, uint64) (types.Transaction, uint8, error)
+
 func DecodeTx(encodedTx []byte, efficiencyPercentage byte, forkId uint64) (types.Transaction, uint8, error) {
```

Comment:
I'm aware that this is the legacy code, but as a side note:
it's strange to me that `efficiencyPercentage` parameter is behaving as input & output parameter. I don't see the point. :confused:

```suggestion
func DecodeTx(encodedTx []byte, efficiencyPercentage byte, forkId uint64) (types.Transaction, uint8, error) {
```

Reply:
I think it is used to encode the transaction, but agreed it may be necessary as part of the output. I will leave it as it is for now.